File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        md = {'(': ')', '{': '}', '[':']'}
        l = ['(', '{', '[']
        r = [')', '}', ']']
        for i in range(len(s)):
            if s[i] in l:
                stack.append(s[i])
            else:
                if len(stack) == 0:
                    return False
                el = stack.pop()
                if md[el] != s[i]:
                    return False
        return True if not stack else False
        

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-4.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
       vector<int> res;
       if (!root) return res;
       queue<TreeNode*> q;
       q.push(root);

       while (!q.empty()) {
        TreeNode* curr;
        int size = q.size();
        for (int _ = 0; _ < size; _++) {
            curr = q.front(); q.pop();
            if (curr->left != nullptr) q.push(curr->left);
            if (curr->right != nullptr) q.push(curr->right);
        }
        res.push_back(curr->val);
       } 
       return res;
    }
};
